systems-ui/src/common/components/systems/system-overview/SystemOverview.js

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import classnames from 'classnames';
+import styles from './SystemOverview.css';
+import {formatDateByTimestamp} from "../../../utils/date/date";
+import {getBorderClassByValue, getStateIdentifier} from "../../../business/systemsState";
+import {systemType} from "../../../types/systemsTypes";
+import { Link } from 'react-router-dom'
+
+const SystemOverview = ({system}) => {
+  const stateClass = getBorderClassByValue(system[getStateIdentifier()]);
```

Comment:
Would this component benefit from your point of view having a `state` to save possible changing values like `stateClass`, `last_boot` and `modified` to let the component be re-rendered only in case of a change of those values?

Reply:
The render of this component is so small that probably adding a PureComponent here (only rerender when the props or state change) would probably be a early optimisation and have even worst performance. We have to take in consideration here that the react render is really optimised when it doesn't change anything in the dom.

I think the react team really optimised the functional stateless components in the new version 16 because it doesn't have all the component lifecycle overhead. Adding all this overhead to just define a shouldComponentUpdate function and run that shallowComparation every time would be more costly than just run the render again.

My rule regarding PureComponent has been: don't use it until you really need it, there are some cases that is obvious, but it's a good practise to measure the perfomance before using it to really check if it changed anything